webrender/res/ps_blend.glsl

```diff
@@ -116,7 +116,10 @@ vec4 HueRotate(vec4 Cs, float amount) {
 }
 
 vec4 Invert(vec4 Cs, float amount) {
-    return mix(Cs, vec4(1.0, 1.0, 1.0, Cs.a) - vec4(Cs.rgb, 0.0), amount);
+    vec4 color = mix(Cs, vec4(Cs.a) - vec4(Cs.rgb, 0.0), amount);
+
+    // Pre-multiply the alpha into the output value.
+    return vec4(color.rgb * clamp(color.a, 0.0, 1.0), color.a);
```

Comment:
`color` is already premultiplied; here you're multiplying it a second time. Your yaml reftest doesn't catch it because it uses a black result, and multiplying black one more time still stays black.

Reply:
Ah, I guess we don't have an (enabled) gecko reftest that catches that either. Will add a specific reftest for this to wrench.